### PR TITLE
add `DateTimeUtcFromDate` schema

### PR DESCRIPTION
**Pull request by wewelll**
## Type

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related

- Related Issue #
- Closes #

### CODE REVIEW

File: packages/effect/src/Schema.ts [hunk 1]
```diff
@@ -6209,6 +6209,21 @@ export class DateTimeUtcFromNumber extends transformOrFail(
...
+ *
+ * @category DateTime.Utc transformations
+ */
+export class DateTimeUtcFromDate extends transformOrFail(
+  Date$.annotations({ description: "a date that will be parsed into a DateTime.Utc" }),
+  DateTimeUtcFromSelf,
+  {
+    strict: true,
+    decode: decodeDateTime,
+    encode: (dt) => ParseResult.succeed(dateTime.toDate(dt))
```
Comment (tim-smart) [inline packages/effect/src/Schema.ts:6223]:
```suggestion
    encode: (dt) => ParseResult.succeed(dateTime.toDateUtc(dt))
```

File: packages/effect/src/Schema.ts [hunk 2]
```diff
@@ -6353,6 +6368,25 @@ export class DateTimeZonedFromSelf extends declare(
...
+  Date$.annotations({ description: "a Date that will be parsed into a DateTime.Zoned" }),
+  DateTimeZonedFromSelf,
+  {
+    strict: true,
+    decode: (s, _, ast) =>
+      option_.match(dateTime.makeZoned(s), {
+        onNone: () => ParseResult.fail(new ParseResult.Type(ast, s)),
+        onSome: ParseResult.succeed
+      }),
+    encode: (dt) => ParseResult.succeed(dateTime.toDate(dt))
```
Comment (tim-smart) [inline packages/effect/src/Schema.ts:6386]:
```suggestion
    encode: (dt) => ParseResult.succeed(dateTime.toDateUtc(dt))
```

**Review by tim-smart — CHANGES_REQUESTED** [review]

File: packages/effect/src/Schema.ts (hunk 1 above)
Comment (wewelll) [reply packages/effect/src/Schema.ts]: fixed

File: packages/effect/src/Schema.ts (hunk 2 above)
Comment (wewelll) [reply packages/effect/src/Schema.ts]: fixed

**Comment by tim-smart** [comment]
If you add some tests we can merge this :)

File: packages/effect/src/Schema.ts [hunk 3]
```diff
@@ -6209,6 +6209,22 @@ export class DateTimeUtcFromNumber extends transformOrFail(
...
 ).annotations({ identifier: "DateTimeUtcFromNumber" }) {}
 
+/**
+ * Defines a schema that attempts to convert a `Date` to a `DateTime.Utc` instance using the `DateTime.unsafeMake` constructor.
+ *
+ * @category DateTime.Utc transformations
+ * @since 3.11.0
+ */
+export class DateTimeUtcFromDate extends transformOrFail(
+  Date$.annotations({ description: "a date that will be parsed into a DateTime.Utc" }),
```
Comment (sukovanej) [inline packages/effect/src/Schema.ts:6219]: I'd rather expect `DateFromSelf`

Comment (wewelll) [reply packages/effect/src/Schema.ts]: yes, I even used `ValidDateFromSelf` because the Date must always be valid anyway

**Comment by wewelll** [comment]
@tim-smart 
I replaced the transformation encoded side with `ValidDateFromSelf`, and added tests !

File: packages/effect/src/Schema.ts [hunk 4]
```diff
@@ -6311,6 +6311,22 @@ export class DateTimeUtcFromNumber extends transformOrFail(
...
 ).annotations({ identifier: "DateTimeUtcFromNumber" }) {}
 
+/**
+ * Defines a schema that attempts to convert a valid `Date` to a `DateTime.Utc` instance using the `DateTime.unsafeMake` constructor.
+ *
+ * @category DateTime.Utc transformations
+ * @since 3.11.0
+ */
+export class DateTimeUtcFromDate extends transformOrFail(
+  ValidDateFromSelf.annotations({ description: "a valid Date that will be parsed into a DateTime.Utc" }),
```
Comment (tim-smart) [inline packages/effect/src/Schema.ts:6321]: You can use DateFromSelf here as `decodeDateTime` will validate the Date instance.

File: packages/effect/src/Schema.ts [hunk 5]
```diff
@@ -6455,6 +6471,26 @@ export class DateTimeZonedFromSelf extends declare(
   }
 ) {}
 
+/**
+ * Defines a schema that attempts to convert a valid `Date` to a `DateTime.Zoned` instance.
+ *
+ * @category DateTime.Zoned transformations
+ * @since 3.11.0
+ */
+export class DateTimeZonedFromDate extends transformOrFail(
```
Comment (tim-smart) [inline packages/effect/src/Schema.ts:6480]: I think we should remove this as a Date can only contain the local timezone.

**Review by tim-smart — CHANGES_REQUESTED** [review]

File: packages/effect/src/Schema.ts (hunk 4 above)
Comment (wewelll) [reply packages/effect/src/Schema.ts]: done

**Review by tim-smart — APPROVED** [review]

**Review by gcanti — CHANGES_REQUESTED** [review]

File: packages/effect/src/Schema.ts [hunk 6]
```diff
@@ -6359,6 +6359,22 @@ export class DateTimeUtcFromNumber extends transformOrFail(
   }
 ).annotations({ identifier: "DateTimeUtcFromNumber" }) {}
 
+/**
+ * Defines a schema that attempts to convert a `Date` to a `DateTime.Utc` instance using the `DateTime.unsafeMake` constructor.
+ *
+ * @category DateTime.Utc transformations
+ * @since 3.11.0
```
[the remark this answers is not in the document]
Comment (wewelll) [reply packages/effect/src/Schema.ts]: fixed

**Review by gcanti — APPROVED** [review]

**Comment by tim-smart** [comment]
🚀

**Pull request merged**